### PR TITLE
Add standard message types that will be used inside templates

### PR DESCRIPTION
**Pull request by guptasu**
Add standard message types that will be used inside templates to represent fields that are

IP_ADDRESS, URI, DURATION, TIMESTAMP, EMAIl_ADDRESS, DNS_NAME

**NOTE:** Currently we have duration and timestamp already covered with google.protobuf.* types, but for consistency we should replace them too with types inside standard_type.proto

### CODE REVIEW

File: mixer/v1/config/descriptor/standard_type.proto [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+message DNSName {}
+
+// EmailAddress is used inside templates for fields that are of ValueType "EMAIL_ADDRESS"
+message EmailAddress {}
+
+// Uri is used inside templates for fields that are of ValueType "URI"
+message Uri {}
+
+// Duration is used inside templates for fields that are of ValueType "DURATION"
+message Duration {}
```
Comment (geeknoid) [inline mixer/v1/config/descriptor/standard_type.proto]: Don't we already supported this by using the normal Timestamp and Duration types?

File: mixer/v1/config/descriptor/standard_type.proto [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 Istio Authors
```
Comment (geeknoid) [inline mixer/v1/config/descriptor/standard_type.proto]: Maybe standard_types.proto would be better (plural form)?

File: mixer/v1/config/descriptor/standard_type.proto (hunk 1 above)
Comment (guptasu) [reply mixer/v1/config/descriptor/standard_type.proto]: Yes, but I think there is no benefit to keep it that way.. everything in template can be expressed in a self contained type system as oppose to a mix of standard_types and google.protobug.*.. From standard_types we generate nice types in `Go` and `proto`; for example for standard_types#Duration we will generate google.protobuf.Duration and time.Duration inside the generated files. 

I mainly added those two types mainly to achieve consistency.

Comment (geeknoid) [reply mixer/v1/config/descriptor/standard_type.proto]: OK, I buy that.

**Comment by guptasu** [comment]
@geeknoid alright, the types are now moved to /template dir..

the usage would look something like:
`istio.mixer.v1.template.IPAddress source_ip = 1;`

**Pull request merged**